dash-renderer/src/actions/constants.js

```diff
@@ -1,18 +1,18 @@
 const actionList = {
-    ON_PROP_CHANGE: 'ON_PROP_CHANGE',
-    SET_REQUEST_QUEUE: 'SET_REQUEST_QUEUE',
-    COMPUTE_GRAPHS: 'COMPUTE_GRAPHS',
-    COMPUTE_PATHS: 'COMPUTE_PATHS',
-    SET_LAYOUT: 'SET_LAYOUT',
-    SET_APP_LIFECYCLE: 'SET_APP_LIFECYCLE',
-    SET_CONFIG: 'SET_CONFIG',
-    ON_ERROR: 'ON_ERROR',
-    SET_HOOKS: 'SET_HOOKS',
+    ON_PROP_CHANGE: 1,
+    SET_REQUEST_QUEUE: 1,
+    SET_GRAPHS: 1,
+    SET_PATHS: 1,
+    SET_LAYOUT: 1,
+    SET_APP_LIFECYCLE: 1,
+    SET_CONFIG: 1,
+    ON_ERROR: 1,
+    SET_HOOKS: 1,
 };
 
 export const getAction = action => {
     if (actionList[action]) {
-        return actionList[action];
+        return action;
```

Comment:
Minor point, `actionList.hasOwnProperty(action)` is supported since IE 5.5+, does the same and is actually slightly faster on Chrome (100ms for 1e6 iterations vs. 130ms) and both implementations have identical performance on FF.

Reply:
Do we still find `getAction` valuable? (cc @chriddyp) Its whole point, I guess, is to make debugging easier by throwing an error if an unrecognized action is dispatched, rather than just silently doing nothing. But we haven't been following that pattern particularly, and our project doesn't lend itself to new actions being added very often anyway, which seems like it's where this would come in handy. I'm inclined to just take `getAction` out entirely.

Reply:
I always found the debugging helpful, I've wasted many hours without it. But I agree that not many actions are being added lately.

Reply:
@Marc-Andre-Rivet points out that, of course, this is exactly the kind of issue TypeScript was made for - keep all the overhead and debugging in the compile step 😁 
Anyway for now I'll leave `getAction` in place but I'm not going to invest in applying the pattern consistently.